core/src/main/java/org/bitcoinj/core/BloomFilter.java

```diff
@@ -126,6 +126,8 @@ public BloomFilter(int elements, double falsePositiveRate, long randomNonce, Blo
         // Optimal number of hash functions for a given filter size and element count.
         hashFuncs = (int)(data.length * 8 / (double)elements * log(2));
         hashFuncs = max(1, min(hashFuncs, MAX_HASH_FUNCS));
+        checkArgument(randomNonce >= 0 && randomNonce <= ByteUtils.MAX_UNSIGNED_INTEGER, () ->
+                "nonce out of range: " + randomNonce);
```

Comment:
Rather than checking here, we should just use an `int`.

Reply:
Agreed, as this is arbitrary data.